src/notation/internal/notationactioncontroller.cpp

```diff
@@ -1907,6 +1920,110 @@ void NotationActionController::toggleStrike()
     currentNotationInteraction()->toggleStrike();
 }
 
+bool NotationActionController::isMasterNotation(const INotationPtr notation) const
+{
+    return globalContext()->currentMasterNotation()->notation() == notation;
+}
+
+QList<INotationPtr> NotationActionController::getOpenNotations()
+{
+    QList<INotationPtr> notations;
+    auto excerpts = globalContext()->currentMasterNotation()->excerpts().val;
+    for (IExcerptNotationPtr excerpt: excerpts) {
+        INotationPtr notation = excerpt->notation();
+        if (notation->opened().val) {
+            notations.append(notation);
+        }
+    }
+    return notations;
+}
+
+void NotationActionController::closeCurrentNotation()
+{
+    INotationPtr nextNotation = nullptr;
+    IMasterNotationPtr masterNotation = globalContext()->currentMasterNotation();
+    QList<INotationPtr> notations = getOpenNotations();
+    for (int i = 0; i < notations.size(); i++) {
+        INotationPtr notation = notations[i];
+        if (notation == currentNotation()) {
+            if (i == notations.size() - 1) { //If last element in list
+                if (i != 0) { //Last element, therefore open part to the left of current part
+                    nextNotation = notations[i - 1];
+                } else { //If element is first one, then there is no more parts to open, so open main project notation
+                    nextNotation = masterNotation->notation();
+                }
+            } else {
+                nextNotation = notations[i + 1];
+            }
+            break;
+        }
+    }
+    if (isMasterNotation(currentNotation())) {
+        dispatcher()->dispatch("file-close");
+    } else {
+        currentNotation()->setOpened(false);
+    }
+    if (nextNotation) {
+        globalContext()->setCurrentNotation(nextNotation);
+    }
+}
+
+void NotationActionController::openPreviousNotation()
+{
+    QList<INotationPtr> notations = getOpenNotations();
+    INotationPtr previousNotation = nullptr;
+    if (globalContext()->currentMasterNotation()->notation() == currentNotation()) { //If the main project is opened, previous part is previous notation
+        previousNotation = notations[notations.size() - 1];
+    } else {
+        for (int i = 0; i < notations.size(); i++) {
+            if (notations[i] == currentNotation()) {
+                if (i != 0) {//If the opened part is not first
+                    previousNotation = notations[i - 1];
+                } else { //If the opened part is first, the next notation is the main project notation
+                    previousNotation = globalContext()->currentMasterNotation()->notation();
+                }
+                break;
+            }
```

Comment:
You have duplicated this logic here and in closeCurrentNotation. Better to make separate methods for getting next / previous notation and use them in close / open methods.
Each method should have one responsibility (S from SOLID https://en.wikipedia.org/wiki/SOLID)
Also we need to implement in such a way that the implementation of the logic can be re-used elsewhere, without having to re-implement it.